worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java

```diff
@@ -132,9 +132,9 @@ private void handleSingleBatchId(final String batchId, final BatchWorkerServices
             final Path pathOfSubBatches = fileSystemProvider.getPathForBatch(tenantId, batchIdExtracted);
 
             if (!Files.exists(pathOfSubBatches)) {
-                log.error("Exception while reading the batch folder: " + pathOfSubBatches + ", the directory does not exist");
-                throw new BatchDefinitionException("Exception while reading the batch folder: " + pathOfSubBatches
-                    + ", the directory does not exist");
+                log.error("Exception while reading the batch: " + pathOfSubBatches + ", it was not found");
+                throw new BatchDefinitionException("Exception while reading the batch: " + pathOfSubBatches
+                    + ", it was not found");
```

Comment:
I'm sorry but this is still too much.  You're exposing the internal path where batches are stored as part of the error message.  This is a security issue.